src/ripple/app/main/NodeStoreScheduler.cpp

```diff
@@ -47,11 +47,14 @@ void NodeStoreScheduler::onChildrenStopped ()
 
 void NodeStoreScheduler::scheduleTask (NodeStore::Task& task)
 {
-    ++m_taskCount;
-    m_jobQueue->addJob (
-        jtWRITE,
-        "NodeObject::store",
-        [this, &task] (Job&) { doTask(task); });
+    if (! isStopping())
+    {
+        ++m_taskCount;
+        m_jobQueue->addJob (
+            jtWRITE,
+            "NodeObject::store",
+            [this, &task] (Job&) { doTask(task); });
+    }
```

Comment:
We need to make sure all writes go to disk. A shutdown after a write causing the write to be lost is not acceptable. If we called this code, it's because we wrote something. When we wrote something, we weren't shut down. We really should figure out what's really going on here and not cause data loss on normal shutdown.